package-system/AWSGameLiftServerSDK/docker_build_linux.sh

```diff
@@ -27,35 +55,64 @@ build_package() {
 
     echo "Generating $1"
 
-    cmake -G Ninja -S ${SRC_PATH} -B $1 -DBUILD_SHARED_LIBS=$2 -DCMAKE_BUILD_TYPE=Release
+    lib_type=$1
+    if [ "$lib_type" == "Shared" ]
+    then
+        build_shared=ON
+    else
+        build_shared=OFF
+    fi
+
+    echo cmake -S ${SRC_PATH} -B ${BUILD_PATH_ROOT}/${lib_type} \
+          -G "Unix Makefiles" \
+          -DBUILD_SHARED_LIBS=$build_shared \
+          -DCMAKE_BUILD_TYPE=Release \
+          -DCMAKE_MODULE_PATH=$DOWNLOADED_PACKAGE_FOLDERS
+
+
+    cmake -S ${SRC_PATH} -B ${BUILD_PATH_ROOT}/${lib_type} \
+          -G "Unix Makefiles" \
+          -DBUILD_SHARED_LIBS=$build_shared \
+          -DCMAKE_BUILD_TYPE=Release \
+          -DCMAKE_MODULE_PATH=$DOWNLOADED_PACKAGE_FOLDERS
```

Comment:
The `set -o xtrace` and `set +o xtrace` options can be used to avoid needing to copy and paste the command that is being executed.

Moreover that can be encapsulated in a bash function
```bash
run_and_trace()
{
    set -o xtrace
    "$@"
    set +o xtrace
}

run_and_trace cmake -S ${SRC_PATH} -B ${BUILD_PATH_ROOT}/${lib_type} \
          -G "Unix Makefiles" \
          -DBUILD_SHARED_LIBS=$build_shared \
          -DCMAKE_BUILD_TYPE=Release \
          -DCMAKE_MODULE_PATH=$DOWNLOADED_PACKAGE_FOLDERS
```

Reply:
actually, using `set -o xtrace` within the script and then `set +o xtrace` will eat up the result of the command since the `set +o xtrace` will reset `?$` to zero every time. I'm going to use a different approach
```
CMD="\
cmake -S ${SRC_PATH} -B ${BUILD_PATH_ROOT}/${lib_type} \
          -G \"Unix Makefiles\" \
          -DBUILD_SHARED_LIBS=$build_shared \
          -DCMAKE_BUILD_TYPE=Release \
          -DCMAKE_MODULE_PATH=$DOWNLOADED_PACKAGE_FOLDERS"
echo $CMD
eval $CMD
if [ $? -ne 0 ] 
...
```

Reply:
You can use a subshell then. with only `set -o xtrace` then.
i.e
```bash
$(set -o xtrace; $CMD)
```

Reply:
That  does not work
```
./build-linux.sh: line 159: Sending: command not found
```